frontend/components/SamplesPage/StepSlider.tsx

```diff
@@ -23,7 +23,30 @@ type StepSliderProps = {
 const StepSlider: FunctionComponent<StepSliderProps> = ({onChange, value, steps}) => {
     const {t} = useTranslation('samples');
     const [step, setStep] = useState(value);
-    useEffect(() => setStep(value), [setStep, value]);
+    const timer = useRef(null as NodeJS.Timeout | null);
+
+    useEffect(() => setStep(value), [value]);
+
+    const clearTimer = useCallback(() => {
+        if (timer.current) {
+            clearTimeout(timer.current);
+            timer.current = null;
+        }
+    }, []);
+
+    // debounce for backend performance consideration
+    useEffect(() => {
+        timer.current = setTimeout(() => {
+            onChange?.(step);
+        }, 500);
+        return clearTimer;
+    }, [step, onChange, clearTimer]);
+
+    // trigger immediately when mouse up
+    const changeComplate = useCallback(() => {
+        clearTimer();
```

Comment:
I would like to wrap these timer things with `useDebounce`

Reply:
There is a usedebounce hook. But it's not satisfied with the situation here. Because timer needs to be cleared in `changeComplate`. So I didn't use the hook.